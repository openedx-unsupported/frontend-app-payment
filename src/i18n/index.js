import { messages as paragonMessages } from '@edx/paragon';
import { messages as headerMessages } from '@edx/frontend-component-header';
import { messages as footerMessages } from '@edx/frontend-component-footer';

import arMessages from './messages/ar.json';
import frMessages from './messages/fr.json';
import es419Messages from './messages/es_419.json';
import es419MessagesProtected from '../subscription/i18n-protected-messages/es_419.json';
import zhcnMessages from './messages/zh_CN.json';
import ptMessages from './messages/pt.json';
import itMessages from './messages/it.json';
import ukMessages from './messages/uk.json';
import deMessages from './messages/de.json';
import ruMessages from './messages/ru.json';
import hiMessages from './messages/hi.json';
import frCAMessages from './messages/fr_CA.json';
import deDEMessages from './messages/de_DE.json';
import itITMessages from './messages/it_IT.json';
import ptPTMessages from './messages/pt_PT.json';
// no need to import en messages-- they are in the defaultMessage field

const appMessages = {
  ar: arMessages,
  // override the protected translations
  'es-419': { ...es419Messages, ...es419MessagesProtected },
  fr: frMessages,
  'zh-cn': zhcnMessages,
  pt: ptMessages,
  it: itMessages,
  de: deMessages,
  hi: hiMessages,
  'fr-ca': frCAMessages,
  ru: ruMessages,
  uk: ukMessages,
  'de-de': deDEMessages,
  'it-it': itITMessages,
  'pt-pt': ptPTMessages,
};

export default [
  headerMessages,
  footerMessages,
  paragonMessages,
  appMessages,
];
