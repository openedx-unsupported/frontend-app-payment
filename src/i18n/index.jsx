import arMessages from './messages/ar.json';
import frMessages from './messages/fr.json';
import es419Messages from './messages/es_419.json';
import es419MessagesProtected from '../subscription/i18n-protected-messages/es_419.json';
import zhcnMessages from './messages/zh_CN.json';
import ptMessages from './messages/pt.json';
import itMessages from './messages/it.json';
import ukMessages from './messages/uk.json';
import deMessages from './messages/de.json';
import ruMessages from './messages/ru.json';
import hiMessages from './messages/hi.json';
import frCAMessages from './messages/fr_CA.json';
import deDEMessages from './messages/de_DE.json';
import itITMessages from './messages/it_IT.json';
import ptPTMessages from './messages/pt_PT.json';
import faMessages from './messages/fa.json';
import faIRMessages from './messages/fa_IR.json';
// no need to import en messages-- they are in the defaultMessage field

const messages = {
  ar: arMessages,
  // override the protected translations
  'es-419': { ...es419Messages, ...es419MessagesProtected },
  fr: frMessages,
  'zh-cn': zhcnMessages,
  pt: ptMessages,
  it: itMessages,
  de: deMessages,
  hi: hiMessages,
  'fr-ca': frCAMessages,
  ru: ruMessages,
  uk: ukMessages,
  'de-de': deDEMessages,
  'it-it': itITMessages,
  'pt-pt': ptPTMessages,
  fa: faMessages,
  'fa-ir': faIRMessages,
};

export default messages;
