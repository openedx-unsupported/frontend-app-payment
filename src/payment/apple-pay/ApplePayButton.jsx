import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { sendTrackEvent } from '@edx/frontend-analytics';
import { performApplePayPayment } from './service';


export default class ApplePayButton extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      canMakePayments: global.ApplePaySession && global.ApplePaySession.canMakePayments(),
    };
  }

  canSubmit() {
    return this.state.canMakePayments
      && !this.props.disabled
      && this.props.totalAmount !== undefined;
  }

  handleClick = () => {
    if (!this.canSubmit()) return;

    // TO DO: after event parity, track data should be
    // sent only if the payment is processed, not on click
    // Check for PayPal and Free Basket as well
    sendTrackEvent(
      'edx.bi.ecommerce.basket.payment_selected',
      {
        type: 'click',
        category: 'checkout',
        paymentMethod: 'Apple Pay',
      },
    );
    performApplePayPayment({
      totalAmount: this.props.totalAmount,
      onPaymentBegin: this.props.onPaymentBegin,
      onPaymentComplete: this.props.onPaymentComplete,
      onMerchantValidationFailure: this.props.onMerchantValidationFailure,
      onPaymentAuthorizationFailure: this.props.onPaymentAuthorizationFailure,
      onPaymentCancel: this.props.onPaymentCancel,
    });
  }

  render() {
    if (!this.state.canMakePayments) return null;

    const other = { ...this.props };

    delete other.totalAmount;
    delete other.onPaymentBegin;
    delete other.onPaymentComplete;
    delete other.onMerchantValidationFailure;
    delete other.onPaymentAuthorizationFailure;
    delete other.onPaymentCancel;

    return (
      <button
        {...other}
        id="applePayBtn"
        onClick={this.handleClick}
        className={classNames('apple-pay-button', this.props.className)}
        lang={this.props.lang}
        title={this.props.title}
      />
    );
  }
}

ApplePayButton.propTypes = {
  totalAmount: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  onPaymentBegin: PropTypes.func,
  onPaymentComplete: PropTypes.func,
  onMerchantValidationFailure: PropTypes.func,
  onPaymentAuthorizationFailure: PropTypes.func,
  onPaymentCancel: PropTypes.func,
  lang: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  className: PropTypes.string,
  disabled: PropTypes.bool,
};

ApplePayButton.defaultProps = {
  totalAmount: undefined,
  onPaymentBegin: undefined,
  onPaymentComplete: undefined,
  onMerchantValidationFailure: undefined,
  onPaymentAuthorizationFailure: undefined,
  onPaymentCancel: undefined,
  className: undefined,
  disabled: false,
};
