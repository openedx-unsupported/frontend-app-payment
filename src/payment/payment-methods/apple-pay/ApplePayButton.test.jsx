import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import { IntlProvider } from '@edx/frontend-platform/i18n';
import { logError } from '@edx/frontend-platform/logging';
import { getConfig } from '@edx/frontend-platform';

import ApplePayButton from './ApplePayButton';

// Mock language cookie
Object.defineProperty(global.document, 'cookie', {
  writable: true,
  value: `${getConfig().LANGUAGE_PREFERENCE_COOKIE_NAME}=en`,
});

jest.mock('@edx/frontend-platform/logging', () => ({
  logError: jest.fn(),
}));

const applePaySession = { begin: jest.fn() };
global.ApplePaySession = jest.fn().mockImplementation(() => applePaySession);
global.ApplePaySession.canMakePayments = () => true;

describe('<ApplePayButton />', () => {
  it('should render properly', () => {
    const tree = renderer
      .create((
        <IntlProvider locale="en">
          <ApplePayButton />
        </IntlProvider>
      ))
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should call an onClick handler if it is supplied', () => {
    const clickHandler = jest.fn();
    const wrapper = mount((
      <IntlProvider locale="en">
        <ApplePayButton onClick={clickHandler} />
      </IntlProvider>
    ));

    wrapper.find('ApplePayButton').simulate('click');
    expect(clickHandler).toHaveBeenCalled();
  });

  it('should not call an onClick handler if it is disabled', () => {
    const clickHandler = jest.fn();
    const wrapper = mount((
      <IntlProvider locale="en">
        <ApplePayButton onClick={clickHandler} disabled />
      </IntlProvider>
    ));

    wrapper.find('ApplePayButton').simulate('click');
    expect(clickHandler).not.toHaveBeenCalled();
  });

  describe('with exploding ApplePaySession canMakePayments', () => {
    let error;

    beforeEach(() => {
      error = new Error();
      global.ApplePaySession.canMakePayments = () => {
        throw error;
      };
    });

    afterEach(() => {
      global.ApplePaySession.canMakePayments = () => true;
    });

    it('should log an error if it catches one in constructor', () => {
      mount((
        <IntlProvider locale="en">
          <ApplePayButton />
        </IntlProvider>
      ));
      expect(logError).toHaveBeenCalledWith(error);
    });
  });
});
