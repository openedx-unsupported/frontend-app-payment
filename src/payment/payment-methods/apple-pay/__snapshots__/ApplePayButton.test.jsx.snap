// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ApplePayButton /> should render properly 1`] = `
<button
  className="apple-pay-button"
  disabled={false}
  id="applePayBtn"
  lang="en"
  title="Pay with Apple Pay"
  type="button"
/>
`;
